tests/cases/fourslash/codeFixAddMissingParam13.ts

```diff
@@ -0,0 +1,20 @@
+/// <reference path="fourslash.ts" />
+
+////function f(a: string): string;
+////function f(a: string, b: number): string;
+////function f(a: string, b?: number): string {
+////    return "";
+////}
+////f("", 1, "");
+
+verify.codeFix({
+    description: [ts.Diagnostics.Add_missing_parameter_to_0.message, "f"],
+    index: 0,
+    newFileContent:
+`function f(a: string): string;
+function f(a: string, b: number, p0: string): string;
+function f(a: string, b?: number, p0?: string): string {
```

Comment:
Looks like the rule is: Adds `p0` to both of the longest overloads, matching optionality.

Reply:
Reading the code: no, it's the longest overload and the implementation signature, so long as they both have the same length. What about when there are two overloads with the same length as the implementation signature? Should they both get the new parameter? It seems little odd to me, but it seems odder to choose between them, so maybe 'both' is the right answer.

Reply:
The Quick Fix attempts to find and modify the longest overload and signature declaration. I thought it might seem odd to only change the overload. Would it be acceptable to modify the overload?

Reply:
What I meant was, what about the case when there are two overloads with the same length?

```ts
function f(a: string, b: string): string
function f(a: number, b: number): number
function f(a: number | string, b: number | string): number | string {
  return a + b
}
f(1, 2, "commentary: adding 1 and 2")
```

should add a new parameter to all 3 signatures.

(Aside: this would be a great quick fix to follow up with AI-powered rename, since it should obviously be named `comment` or `message` but Typescript has no way of calculating that.)

Reply:
Oke, I've added the [test](https://github.com/microsoft/TypeScript/pull/56411/files#diff-370f33a81a61409d162a7737ad0cbaf02ab3a7ef6c16746b5eaa8fc76452863cR1-R20) to cover this case